peachjam/resources.py

```diff
@@ -477,3 +480,91 @@ def after_import_row(self, row, instance, row_number=None, **kwargs):
                 self.download_attachment(
                     row["media_summary_file"], judgment, "Media summary"
                 )
+
+
+class ArticleAuthorWidget(ForeignKeyWidget):
+    def clean(self, value, row=None, *args, **kwargs):
+        if not value:
+            raise ValidationError("author is required")
+        author = value
+        first_name, *last_name = value.split()
+        author, _ = self.model.objects.get_or_create(
+            username=slugify(value),
+            defaults={
+                "first_name": first_name,
+                "last_name": last_name[0],
+            },
+        )
+
+        return author
+
+
+class ImageWidget(CharWidget):
+    def clean(self, value, row=None, *args, **kwargs):
+        try:
+            file_url = value
+            file = download_source_file(file_url)
+            mime, _ = mimetypes.guess_type(file_url)
+            file_ext = mimetypes.guess_extension(mime)
+            return File(file, name=f"{slugify(value[:20])}{file_ext}")
+        except requests.exceptions.RequestException as e:
+            msg = getattr(e, "message", repr(e)) or repr(e)
+            logger.warning(f"Error downloading image: {value} - {msg}")
+            raise ValidationError(f"Error downloading image: {value} - {msg}")
+
+
+class PublishedWidget(BooleanWidget):
+    def clean(self, value, row=None, **kwargs):
+        if value:
+            return True
+        return False
+
+
+class TopicsWidget(ManyToManyWidget):
+    def clean(self, value, row=None, **kwargs):
+        if value:
+            article_tag_root = Taxonomy.objects.filter(
+                name__iexact="Article tags"
+            ).first()
+            if not article_tag_root:
```

Comment:
I have created a root taxonomy called Article Tag just for articles.

Reply:
@longhotsummer Creating a new root taxonomy creates a new listing on the taxonomy page like this:
![image](https://user-images.githubusercontent.com/25079238/228166346-15375999-04c2-4401-bf94-b2e5eddecaa1.png)

Articles do not appear in the listings though

Reply:
That's fine for now, I think. We can finesse it later. At least this way the information is not lost.